src/bin/pg_autoctl/cli_formation.c

```diff
@@ -199,7 +201,21 @@ keeper_cli_formation_create_getopts(int argc, char **argv)
 				log_trace("--disable-secondary");
 				break;
 			}
-
+			case 'r':
+			{
+				/* { "number-sync-stanbys", required_argument, NULL, 'S'} */
+				int numberSyncStanbys = strtol(optarg, NULL, 10);
+
+				if (errno == EINVAL|| numberSyncStanbys < 0)
+				{
+					log_fatal("--number-sync-stanbys argument is not valid."
+							  " Use a non-negative integer value.");
```

Comment:
I picked non-negative because we also support 0 there. Shall we drop supporting 0 ?

Reply:
I checked citus and PG code for similar usage. Found that
PG has following snippet in  `src/backend/partitioning/partbounds.c` function `satisfies_hash_partition()`
```c
	if (remainder < 0)
		ereport(ERROR,
				(errcode(ERRCODE_INVALID_PARAMETER_VALUE),
				 errmsg("remainder for hash partition must be a non-negative integer")));
```

So I am inclined to keep the message this way

Reply:
I completely overlooked that you can't provide a negative number to cli command easily
when you wrote `pg_autoctl set node candidate-priority -1 --pgdata primary `  parameter parser treats -1 as a parameter and can't find it in valid options and we fail way before in the cli.

Reply:
What about `-- candidate-priority -1` where the `--` usage signals to getup that there's no options to parse after that? Otherwise, agreed, let's use “non-negative” as in Citus, it might be the best way to have an easy to understand error message.